Cache intelligence v2 (#618)
harness/developer-hub: docs/continuous-integration/troubleshoot/optimizing-ci-build-times.md

> ### Cache and Reuse the Data for Your Fetch Operations

Reviewer: ```suggestion
### Cache and reuse the data for your fetch operations
```

> Caching and reusing can be useful for data that your builds need to fetch, but that you cannot include in an optimized image as described above. Caching ensures faster job execution by reusing the expensive fetch operation data from previous jobs. 

Reviewer: ```suggestion
Caching and reusing is useful for data that your builds need to fetch, but that you can't include in an optimized image as described above. Caching expedites job execution by reusing expensive fetch operation data from previous jobs. 
```
